src/prop/theory_proxy.cpp

```diff
@@ -162,15 +171,32 @@ void TheoryProxy::notifySkolemDefinition(Node a, TNode skolem)
   d_skdm->notifySkolemDefinition(skolem, a);
 }
 
-void TheoryProxy::notifyAssertion(Node a, TNode skolem, bool isLemma)
+void TheoryProxy::notifyAssertion(Node a,
+                                  TNode skolem,
+                                  bool isLemma,
+                                  bool local)
 {
   // ignore constants
   if (a.isConst())
   {
     return;
   }
   // notify the decision engine
-  d_decisionEngine->addAssertion(a, skolem, isLemma);
+  std::vector<TNode> assertions{a};
+  if (local)
+  {
+    // If it is marked local, add as local assertions.
+    d_decisionEngine->addLocalAssertions(assertions);
```

Comment:
```suggestion
    d_decisionEngine->addLocalAssertions({a});
```

Reply:
Avoids unnecessary creation of assertions if none of the branches are triggered.